src/prpy/base/mobilebase.py

```diff
@@ -85,20 +85,17 @@ def Forward(self, meters, execute=True, direction=None, **kw_args):
         if abs(numpy.linalg.norm(direction) - 1.0) > 1e-3:
             raise ValueError('Direction must be a unit vector.')
 
-        if self.simulated:
-            with self.robot.GetEnv():
-                start_pose = self.robot.GetTransform()
-                offset_pose = numpy.eye(4)
-                offset_pose[0:3, 3] = meters * direction
-                goal_pose = numpy.dot(start_pose, offset_pose)
-
-            path = create_affine_trajectory(self.robot, [ start_pose, goal_pose ])
-            if execute:
-                return self.robot.ExecutePath(path, **kw_args)
-            else:
-                return path
+        with self.robot.GetEnv():
+            start_pose = self.robot.GetTransform()
+            offset_pose = numpy.eye(4)
+            offset_pose[0:3, 3] = meters * direction
+            goal_pose = numpy.dot(start_pose, offset_pose)
+            
+        path = create_affine_trajectory(self.robot, [ start_pose, goal_pose ])
+        if execute:
+            return self.robot.ExecutePath(path, **kw_args)
         else:
-            raise NotImplementedError('DriveForward is not implemented')
+            return path
```

Comment:
I think this is something we want to keep.  

If I recall, previously we were raising the error in prpy because we might not always be capable of executing these trajectories on the real robot. Then in herbpy, we were overriding this function to actually return the path, because we know with herb we can execute them.

This flips that thinking. Basically, we assume all robots can execute these affine trajectories. If a robot can't, it needs to override this function.  I think this also matches the thinking of the regular planning pipeline.